system/logaccess/ElasticStack/ElasticStackLogAccess.cpp

```diff
@@ -0,0 +1,502 @@
+{
+    try
+    {
+        LOG(reqloglevel,"ESLogAccess: Requesting '%s'... ", logmessageprefix );
+        cpr::Response esREsponse = m_esClient.performRequest(httpmethod,url,reqbody);
+        Owned<IPropertyTree> response = createPTreeFromJSONString(esREsponse.text.c_str());
+        LOG(resploglevel,"ESLogAccess: '%s' response: '%s'", logmessageprefix, esREsponse.text.c_str());
+
+        return response.getClear();
+    }
+    catch (ConnectionException & ce)//std::runtime_error
+    {
+        LOG(MCuserError, "ESLogAccess: Encountered error requesting '%s': '%s'", logmessageprefix, ce.what());
+    }
+    catch (...)
+    {
+        LOG(MCuserError, "ESLogAccess: Encountered error requesting '%s'", logmessageprefix);
+    }
+    return nullptr;
+
+}
+
+const IPropertyTree * ElasticStackLogAccess::getTimestampTypeFormat(const char * indexpattern, const char * fieldname)
+{
+    if (isEmptyString(indexpattern))
+        throw makeStringException(-1, "ElasticStackLogAccess::getTimestampTypeFormat: indexpattern must be provided");
+
+    if (isEmptyString(fieldname))
+        throw makeStringException(-1, "ElasticStackLogAccess::getTimestampTypeFormat: fieldname must be provided");
+
+    VStringBuffer timestampformatreq("%s/_mapping/field/created_ts?include_type_name=true&format=JSON", indexpattern);
+    return performAndLogESRequest(Client::HTTPMethod::GET, timestampformatreq.str(), "", "getTimestampTypeFormat");
+}
+
+const IPropertyTree * ElasticStackLogAccess::getIndexSearchStatus(const char * indexpattern)
+{
+    if (!indexpattern || !*indexpattern)
+        throw makeStringException(-1, "ElasticStackLogAccess::getIndexSearchStatus: indexpattern must be provided");
+
+    VStringBuffer indexsearch("_cat/indices/%s?format=JSON", indexpattern);
+    return performAndLogESRequest(Client::HTTPMethod::GET, indexsearch.str(), "", "List of available indexes");
+
+}
+
+const IPropertyTree * ElasticStackLogAccess::getESStatus()
+{
+    return performAndLogESRequest(Client::HTTPMethod::GET, "_cluster/health", "", "Target cluster health");
+}
+
+/*
+ * Transform ES query response to back-end agnostic response
+ *
+ */
+void processESJsonResp(const cpr::Response & retrievedDocument, StringBuffer & returnbuf, LogAccessLogFormat format)
+{
+    if (retrievedDocument.status_code != 200)
+        throw makeStringExceptionV(-1, "ElasticSearch request failed: %s", retrievedDocument.status_line.c_str());
+
+    DBGLOG("Retrieved ES JSON DOC: %s", retrievedDocument.text.c_str());
+
+    Owned<IPropertyTree> tree = createPTreeFromJSONString(retrievedDocument.text.c_str());
+    if (!tree)
+        throw makeStringExceptionV(-1, "%s: Could not parse ElasticSearch query response", COMPONENT_NAME);
+
+    if (tree->hasProp("timed_out") && tree->getPropBool("timed_out"))
```

Comment:
note: the "tree->hasProp("timed_out")" can be removed from this line and line 158 and have the same effect.  Even clearer if a default of false/0 is passed to the getProp function.